### DISCUSSION TITLE
Renovate to support settings --base-branches via cli?

### DISCUSSION THREAD

**Discussion by smallgiantbloo** [renovatebot/renovate#32505] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab - 39.10.0

### Please tell us more about your question or problem

I have three different branches in my repository. 
The default branch where there arent't any dependencies that requires updates but stores the renovate config file.
Other two branches have dependencies that renovate should update. Sometimes I need to update the two branches at the same time and sometimes I would need to update only one branch at the time.

So I have a situation where I would need to trigger Renovate dynamically to different branches.
I would need to override the baseBranches-setting from the default branch.

I'm running the renovate-image in Gitlab.
It could be possible using the `--force-cli` switch and overriding the baseBranches if the cli would support that.

Is there a reason the cli doesn't support setting --base-branches?
https://docs.renovatebot.com/configuration-options/#basebranches

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
baseBranches is a per-repo setting. It's very rare that any user/org has the exact same baseBranches for all repos, so it does not make sense to allow it to be globally configurable.

Instead you could JSON stringify your config and use `RENOVATE_CONFIG` in env, for example.

**Reply by smallgiantbloo** [reply]
thanks,

probably many organisations have quite strict version control guides and using the same base branches wouldn't be that unusual.
But also I'm kind of solving the problem of not 100% of the time using the same basebraches value. 
We are using the renovate docker image and parametrizating the renovate runs (with values that override our defaults) are quite handy in that scenario.

**Reply by rarkins** [reply]
You'll need to turn off pruneStaleBranches too if you go this way

**Reply by smallgiantbloo** [reply]
true, setting pruneStaleBranches off is a great heads up! thanks
